### PR TITLE
feat: run Whitebox batch tools from a selected input directory

### LINKED ISSUE #140: Desktop very buggy
I wasn't on for 5 minutes and kept running into road blocks.  I added a copc via aws link and couldn't run the classify tool.  I downloaded the file locally, and the software couldn't find in the path. I converted to .las and it still wouldn't find it.  When I went to browse for it to add the files were grayed-out and not selectable to add to the tool.

This looks like it has great potential.  I'd really like to use it with streaming COPC files.

FYI, I'm on a Mac-Studio

### PR DESCRIPTION
**Pull request by giswqs**
## Summary
- Whitebox LiDAR tools document a batch mode that processes all files in the current directory when the input is omitted. When the user selects a directory as the input, the input argument is now omitted and the tool subprocess runs with that directory as its working directory, enabling batch processing from the UI.
- Default output paths are skipped in batch mode so Whitebox writes outputs next to the inputs.
- Update the demo project URL in the README and docs to a share.geolibre.app example.

## Test plan
- [x] Added unit tests covering directory inputs (batch working directory, no input arg) and file inputs (unchanged behavior, default output path)
- [x] `pytest backend/geolibre_server/tests/` passes
- [x] `pre-commit run --all-files` passes
- [ ] Run a batch-capable Whitebox LiDAR tool with a directory input from the UI and verify outputs are written alongside the inputs

## Summary by CodeRabbit

* **New Features**
  * Added batch-mode support for Whitebox tools, allowing runs to use a selected working directory for tool execution.

* **Documentation**
  * Updated embedded demo examples and browser instructions with a new example project URL and revised parameter examples.

* **Tests**
  * Added tests covering batch-mode argument preparation, detection, conflict handling, and related edge cases.

### CODE REVIEW

File: backend/geolibre_server/geolibre_server/app/whitebox.py [hunk 1]
```diff
@@ -718,6 +720,45 @@ def _coerce_value(value: Any, kind: str) -> Any:
...
+    Returns:
+        The selected directory path when the value should trigger batch mode,
+        otherwise None.
+    """
+    if not _is_batch_directory_input(param) or not isinstance(value, str):
+        return None
+    path = Path(value).expanduser()
+    if path.is_dir():
+        return str(path)
+    return None
```
[the remark this answers is not in the document]
Comment (giswqs) [reply backend/geolibre_server/geolibre_server/app/whitebox.py]: Done in 3d8bf94 — the batch directory now must be absolute and non-symlinked (matching the whitebox_output symlink policy) and is normalized with resolve(). On the allowlist: input and output file paths in this API are already arbitrary user-selected local paths (this is a local desktop backend), so restricting the batch cwd to an allowlist would break the feature without changing the trust model. The /output allowlist exists because that endpoint reads file contents back to the client, which this path does not.

File: backend/geolibre_server/geolibre_server/app/whitebox.py [hunk 2]
```diff
@@ -718,6 +720,45 @@ def _coerce_value(value: Any, kind: str) -> Any:
...
+        as batch-capable when omitted.
+    """
+    kind = str(param.get("kind") or "")
+    description = str(param.get("description") or "").lower()
+    return (
+        kind.endswith("_in")
+        and "batch mode" in description
+        and "current directory" in description
+        and "omitted" in description
+    )
```
[the remark this answers is not in the document]
Comment (giswqs) [reply backend/geolibre_server/geolibre_server/app/whitebox.py:744]: Done in 3d8bf94 — the expected catalog wording is now a documented module constant (_BATCH_DESCRIPTION_PHRASES) and a regression test verifies non-batch descriptions and output kinds never match. A catalog-stable batch_capable flag would need changes in the catalog normalization layer, which is out of scope here; if the wording changes, the failure mode is fall-through to normal argument handling, not misexecution.

File: backend/geolibre_server/geolibre_server/app/whitebox.py [hunk 3]
```diff
@@ -739,28 +780,47 @@ def _write_layer_input(param_name: str, layer: dict[str, Any], temp_paths: list[
...
+            continue
         coerced = _coerce_value(value, kind)
         if coerced is not None:
             args[name] = coerced
 
     for name, spec in specs.items():
         kind = str(spec.get("kind") or "")
-        if kind.endswith("_out") and not args.get(name):
+        if kind.endswith("_out") and not args.get(name) and not working_directory:
             args[name] = _default_output_path(request.tool_id, name, kind)
```
[the remark this answers is not in the document]
Comment (giswqs) [reply backend/geolibre_server/geolibre_server/app/whitebox.py:830]: No change needed — as the comment's own trace concludes, user-supplied output paths are added in the first loop and the not working_directory guard only suppresses auto-generated defaults in batch mode, which is intended.

File: backend/geolibre_server/tests/test_whitebox_arguments.py [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    request = WhiteboxRunRequest(
+        tool_id="classify_lidar",
+        parameters={"input": str(tmp_path), "output": ""},
+        tool=_classify_lidar_tool(),
+    )
+
+    args, working_directory = _prepare_arguments(request, [])
+
+    assert args == {}
+    assert working_directory == str(tmp_path)
```
[the remark this answers is not in the document]
Comment (giswqs) [reply backend/geolibre_server/tests/test_whitebox_arguments.py]: Done in 3d8bf94 — added tests for conflicting batch directories (ValueError branch), nonexistent paths falling through, relative paths, tool=None, equivalent directory variants, and non-batch descriptions, plus a comment explaining why args is empty in the batch test.

File: backend/geolibre_server/geolibre_server/app/whitebox.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (giswqs) [reply backend/geolibre_server/geolibre_server/app/whitebox.py]: Done in 3d8bf94 — _batch_working_directory now returns str(path.resolve()), so the conflict comparison operates on normalized paths and lexical variants of the same directory no longer raise. Covered by test_prepare_arguments_accepts_equivalent_batch_directories.
